src/2-spatial-data-preparation.py

```
@@ -281,84 +325,98 @@ def mask(block, input_nodata, template_nodata):
-        
-        with rioxarray.open_rasterio(restrictionPath, chunks = True) as restrictionRaster:
+        outputRestrictionPath = os.path.join(
+            ssimTempDir, "processed_" + os.path.basename(restrictionPath))
+
+        with rioxarray.open_rasterio(restrictionPath, chunks=True) as restrictionRaster:
             # Check for valid crs
             if pd.isnull(restrictionRaster.rio.crs):
-                raise ValueError(print("The restriction rasters has an invalid or unknown CRS. Ensure that the raster has a valid CRS before continuing."))  
+                raise ValueError(print(
+                    "The restriction rasters has an invalid or unknown CRS. Ensure that the raster has a valid CRS before continuing."))
             elif restrictionRaster.rio.crs.is_valid == False:
```

Ruff (0.13.1)

Avoid equality comparisons to `False`; use `not restrictionRaster.rio.crs.is_valid:` for false checks

Replace with `not restrictionRaster.rio.crs.is_valid`

(E712)